### PR TITLE
feature (RoomsController): provide information about all available rooms

### PR DESCRIPTION
**Pull request by TihomirAleksandrov**
Created "Rooms" page, with information about all of the rooms in the guest house.

### CODE REVIEW

File: BulgarskiChiflikGuestHouse.BLL/Model/RoomServiceModel.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        public Guid Id { get; set; }
+
+        public string Number{ get; set; }
+
+        public int Capacity { get; set; }
+
+        public decimal Price { get; set; }
+
+        public string? Description { get; set; }
```
Comment (AlbertEdward) [inline BulgarskiChiflikGuestHouse.BLL/Model/RoomServiceModel.cs]: chore: nullable fields should be below required fields

File: BulgarskiChiflikGuestHouse.BLL/Service/RoomService.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                Description = room.Description,
+                ImagesSources = room.ImagesSources,
+                Amenities = room.Amenities
+            })
+            .ToList();
+
+            return serviceRooms;
+        }
+
+        public async Task<RoomServiceModel> GetRommByIdAsync(Guid roomId)
```
Comment (AlbertEdward) [inline BulgarskiChiflikGuestHouse.BLL/Service/RoomService.cs]: chore: you have typo - "GetRommByIdAsync"

File: BulgarskiChiflikGuestHouse.Web/Controllers/RoomsController.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    public class RoomsController : Controller
+    {
+        private readonly IRoomService _roomService;
+
+        public RoomsController(IRoomService roomService)
+        {
+            _roomService = roomService;
+        }
+
+        public async Task<IActionResult> OurRooms()
```
Comment (AlbertEdward) [inline BulgarskiChiflikGuestHouse.Web/Controllers/RoomsController.cs]: chore: rename controller method name to follow REST rules

File: BulgarskiChiflikGuestHouse.Web/Controllers/RoomsController.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        {
```
Comment (AlbertEdward) [inline BulgarskiChiflikGuestHouse.Web/Controllers/RoomsController.cs:16]: chore: the controller should be as short as possible. Can you remove the mapping from the controller and add it somewhere else?

File: BulgarskiChiflikGuestHouse.Web/Models/RoomViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+		public int Capacity { get; set; }
+
+		public decimal Price { get; set; }
+
+		public List<ImageSourceServiceModel> ImagesSource { get; set; }
+		
+		public ICollection<AmenityServiceModel> Amenities { get; set; } = new HashSet<AmenityServiceModel>();
+
+		public string Description { get; set; }
```
Comment (AlbertEdward) [inline BulgarskiChiflikGuestHouse.Web/Models/RoomViewModel.cs]: chore: collections must be under primitive types

File: BulgarskiChiflikGuestHouse.Web/Views/Rooms/RoomDetails.cshtml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        <button class="carousel-control-prev" type="button" data-bs-target="#carouselExample" data-bs-slide="prev">
+            <span class="carousel-control-prev-icon" aria-hidden="true"></span>
+            <span class="visually-hidden">Previous</span>
+        </button>
+        <button class="carousel-control-next" type="button" data-bs-target="#carouselExample" data-bs-slide="next">
+            <span class="carousel-control-next-icon" aria-hidden="true"></span>
+            <span class="visually-hidden">Next</span>
+        </button>
+    </div>
+    @* <div id="carouselExampleIndicators" class="carousel slide" data-bs-ride="carousel">
```
Comment (AlbertEdward) [inline BulgarskiChiflikGuestHouse.Web/Views/Rooms/RoomDetails.cshtml]: question: why did you comment the code? If you want to use it later, you should add "TODO:" with a description for it. If not, it should be deleted

File: BulgarskiChiflikGuestHouse.DAL/Models/Room.cs [hunk 7]
```diff
@@ -1,19 +1,29 @@
...
+        [Required]
+        public decimal Price { get; set; }
 
-        public Guid AmenityId { get; set; }
+        public string? Description { get; set; }
 
-        public ICollection<Amenity>? Amenities { get; set; }
+        [ForeignKey(nameof(RoomType))]
+        public Guid RoomTypeId { get; set; }
+        public RoomType RoomType { get; set; }
```
Comment (AlbertEdward) [inline BulgarskiChiflikGuestHouse.DAL/Models/Room.cs:24]: chore: add newline

File: BulgarskiChiflikGuestHouse.Web/Controllers/RoomsController.cs (hunk 3 above)
Comment (TihomirAleksandrov) [reply BulgarskiChiflikGuestHouse.Web/Controllers/RoomsController.cs]: Done.

File: BulgarskiChiflikGuestHouse.Web/Controllers/RoomsController.cs (hunk 4 above)
Comment (TihomirAleksandrov) [reply BulgarskiChiflikGuestHouse.Web/Controllers/RoomsController.cs:16]: It's done.

File: BulgarskiChiflikGuestHouse.BLL/Model/RoomServiceModel.cs (hunk 1 above)
Comment (TihomirAleksandrov) [reply BulgarskiChiflikGuestHouse.BLL/Model/RoomServiceModel.cs]: Changed it.

File: BulgarskiChiflikGuestHouse.BLL/Service/RoomService.cs (hunk 2 above)
Comment (TihomirAleksandrov) [reply BulgarskiChiflikGuestHouse.BLL/Service/RoomService.cs]: it's fixed.

File: BulgarskiChiflikGuestHouse.Web/Models/RoomViewModel.cs (hunk 5 above)
Comment (TihomirAleksandrov) [reply BulgarskiChiflikGuestHouse.Web/Models/RoomViewModel.cs]: Done.

File: BulgarskiChiflikGuestHouse.Web/Views/Rooms/RoomDetails.cshtml (hunk 6 above)
Comment (TihomirAleksandrov) [reply BulgarskiChiflikGuestHouse.Web/Views/Rooms/RoomDetails.cshtml]: Commented them.

File: BulgarskiChiflikGuestHouse.DAL/Models/Room.cs (hunk 7 above)
Comment (TihomirAleksandrov) [reply BulgarskiChiflikGuestHouse.DAL/Models/Room.cs:24]: It's added.

**Review by AlbertEdward — APPROVED** [review]

**Pull request merged**